### ISSUE TITLE
How to use OptSpace function in Python? No fit_transform method

### ISSUE THREAD

**Issue by jolespin** [cjm007/DEICODE#46]
I'm trying to use `U,s,V = OptSpace().fit_transform(A_)` but there are no defaults for `OptSpace` and there is also no `fit_transform` method.

How can I use your `DEICODE` Python package without using `qiime2`? 

Side question, is `OptSpace` only intended for aitchison transformed compositional data?

**Comment by mortonjt** [comment]
I don't think there are any python tutorials on how to run optspace - it was designed to run in the command line.

But there are some unittests that may help you run it : see [here](https://github.com/biocore/DEICODE/blob/master/deicode/tests/test_optspace.py#L151)

**Comment by jolespin** [comment]
Thanks, I was basing it off of this tutorial here: 
https://github.com/biocore/DEICODE/blob/master/ipynb/introduction.ipynb

I vaguely remember trying out DEICODE (maybe it was named something different) in the past before it was integrated into QIIME2

**Comment by cameronmartino** [comment]
We/I should add some docs on how to do this. @mortonjt The only problem I see with running optspace in this way is it would bypass the recentering done in the [rpca](https://github.com/biocore/DEICODE/blob/master/deicode/rpca.py#L12) script. However, you could run rpca as a python API and return a skbio  [OrdinationResults](http://scikit-bio.org/docs/0.5.1/generated/generated/skbio.stats.ordination.OrdinationResults.html) and [DistanceMatrix](http://scikit-bio.org/docs/0.5.1/generated/generated/skbio.stats.distance.DistanceMatrix.html).

**Comment by mortonjt** [comment]
Right, basically use the qiime2 function in python.  That can be done via
```python
from deicode.rpca import rpca
ord_res, dist_res = rpca(microbes_biom)
```

**Comment by jolespin** [comment]
Is  `rpca` a preprocessing step for `OptSpace`?  Is it possible to run `OptSpace` with the input as either a pandas dataframe or numpy array?

**Comment by cameronmartino** [comment]
Hi @jolespin, You will still need to import your data into biom format (see example below on how to do that from a data frame) but I added a tutorial on how to do this. You can find that tutorial [here](https://nbviewer.jupyter.org/github/biocore/DEICODE/blob/master/ipynb/tutorials/moving-pictures-standalone-cli-and-api.ipynb). Hope this helps. (**Note**: the biplot function I used is just a helper for that tutorial and is _not_ unit-tested, so use that with caution). Also, the version in that tutorial is not released yet so run `pip install git+https://github.com/biocore/DEICODE` to get the right version installed.

```python
import pandas as pd
from biom import Table

# a table where the index are features (i.e. ASVs) and columns are samples
table_df = pd.read_csv("path/to/table/table.tsv", sep="\t", index_col=0)
# make biom table
bt = Table(table_df.values, table_df.index, table_df.columns)

```
